### PR TITLE
Feature/Added room membership join date

### PR DESCRIPTION
**Pull request by JerryBii**
Created membership endpoints to add new users to rooms, thereby allowing for retrieval of room join date.

### CODE REVIEW

File: api/src/.vscode/settings.json
Comment (deandrebaker) [inline api/src/.vscode/settings.json:2]: Please undo this change. Thanks!

File: api/src/messaging/api_views.py
Comment (deandrebaker) [inline api/src/messaging/api_views.py:46]: Maybe change this name to something that describes what this API view does, which is to add or remove a user from a room

File: api/src/messaging/api_views.py
Comment (deandrebaker) [inline api/src/messaging/api_views.py:1]: Unused. We can delete

File: api/src/messaging/api_views.py
Comment (deandrebaker) [inline api/src/messaging/api_views.py:3]: Unused. We can delete

File: api/src/messaging/api_views.py
Comment (deandrebaker) [inline api/src/messaging/api_views.py:63]: Probably give a better response message, since we didn't actually create a member

File: api/src/messaging/api_views.py
Comment (deandrebaker) [inline api/src/messaging/api_views.py:67]: I think maybe we can use a status 400 instead of 404? Because the room was still found, but the user in the request body was not. I think this is more of a bad request

File: api/src/messaging/api_views.py
Comment (deandrebaker) [inline api/src/messaging/api_views.py:56]: We should handle the case when the room is not found

File: api/src/messaging/serializers.py
Comment (deandrebaker) [inline api/src/messaging/serializers.py:12]: Doesn't look like we use this serializer anywhere. We can remove it and undo the changes to this file

**Review by deandrebaker — CHANGES_REQUESTED** [review]
@JerryBii great work! I added some request changes. I think you should still get someone else to review and approve after making these changes.

File: api/src/.vscode/settings.json
Comment (NihalPotdar) [reply api/src/.vscode/settings.json]: You might have missed adding back `"python.pythonPath": "env\\Scripts\\python.exe"`

File: api/src/messaging/api_views.py
Comment (NihalPotdar) [inline api/src/messaging/api_views.py:71]: I think it might be more descriptive to say `Room not found` here, unless there's other cases where this statement is triggered

File: api/src/messaging/api_views.py
Comment (NihalPotdar) [inline api/src/messaging/api_views.py:74]: We should handle the case where the room may not be found

**Comment by NihalPotdar** [comment]
Sorry, this is kinda nitpicky but can you please squash your commits together and add your ticket number in the resulting commit? Just makes it easier for tracking.

Looks good otherwise, nice work! I will approve the request once ^ is done

File: api/src/messaging/api_views.py
Comment (NihalPotdar) [reply api/src/messaging/api_views.py:74]: Currently the server will run into trouble if we just query for the case where the room isn't found

Something like in the method above could work

**Review by NihalPotdar — CHANGES_REQUESTED** [review]
Sorry, this is kinda nitpicky but can you please squash your commits together and add your ticket number in the resulting commit? Just makes it easier for tracking.

Looks good otherwise, nice work! I will approve the request once ^ is done

File: api/src/messaging/serializers.py
Comment (deandrebaker) [inline api/src/messaging/serializers.py:5]: Add this 1 line back please

File: api/src/messaging/serializers.py
Comment (deandrebaker) [inline api/src/messaging/serializers.py:31]: Add this 1 line back please

File: api/src/messaging/api_views.py
Comment (deandrebaker) [inline api/src/messaging/api_views.py:75]: We should rename roomid to room_id to be consistent with python snakecase naming convention

File: api/src/.vscode/settings.json
Comment (deandrebaker) [inline api/src/.vscode/settings.json:2]: Add this 1 line back please

**Review by deandrebaker — CHANGES_REQUESTED** [review]
@JerryBii looking better! Just missed some more mistakes

**Comment by deandrebaker** [comment]
> 
> 
> Sorry, this is kinda nitpicky but can you please squash your commits together and add your ticket number in the resulting commit? Just makes it easier for tracking.
> 
> Looks good otherwise, nice work! I will approve the request once ^ is done

@NihalPotdar all commits will be squashed when merging to main by default 👍 . But yeah adding the ticket number to the title of the PR will add the ticket number to the merge commit.

File: api/src/messaging/api_views.py
Comment (JerryBii) [reply api/src/messaging/api_views.py:71]: I have a check-in place to see if the room exists from which lines 68 and 69 are executed. Otherwise, line 71 is executed, I can change the error statuses if that would help.

**Comment by JerryBii** [comment]
> @JerryBii looking better! Just missed some more mistakes

Made the requested changes

File: api/src/messaging/api_views.py
Comment (JerryBii) [reply api/src/messaging/api_views.py:68]: I added a try-catch that returns a "detail: not found" message.

**Review by deandrebaker — APPROVED** [review]

**Comment by deandrebaker** [comment]
@JerryBii please don't merge this branch in until PR #449 has been merged in to main. Thanks

**Review by NihalPotdar — DISMISSED** [review]

**Review by NihalPotdar — APPROVED** [review]

**Review by deandrebaker — APPROVED** [review]

**Pull request merged**
